### PR TITLE
diff: add unit test for chunk split

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
add unit test for chunk split

### What is changed and how it works?
1. move code about split to spliter.go
2. use mocksql to create fake data, and then check the chunk is expected or not.
3. fix some corner case

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/diff/chunk.go [hunk 1]
```diff
@@ -332,36 +340,37 @@ func (s *randomSpliter) splitRange(db *sql.DB, chunk *chunkRange, count int, sch
...
-				if useNewColumn {
-					lower = ""
-					lowerSymbol = ""
-				} else {
-					lower = splitValues[i]
-					lowerSymbol = symbolMin
-				}
+		}
+
+		if i == 0 && valueCounts[i] == 1 {
```
Comment (IANTHEREAL) [inline pkg/diff/chunk.go]: it's very unfriendly to reviewer, please refine these code, make the logic more clear

File: pkg/diff/spliter_test.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+import (
+	"fmt"
+
+	sqlmock "github.com/DATA-DOG/go-sqlmock"
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+)
+
+var _ = Suite(&testSpliterSuite{})
```
Comment (IANTHEREAL) [inline pkg/diff/spliter_test.go:24]: Good Job!

**Comment by WangXiangUSTC** [comment]
@GregoryIan PTAL again

File: Makefile [hunk 3]
```diff
@@ -55,9 +57,7 @@ fmt:
 check:
 	#go get github.com/golang/lint/golint
 	@echo "vet"
-	@ go tool vet $(FILES) 2>&1 | awk '{print} END{if(NR>0) {exit 1}}'
-	@echo "vet --shadow"
-	@ go tool vet --shadow $(FILES) 2>&1 | awk '{print} END{if(NR>0) {exit 1}}'
+	$(GO) vet -all $(PACKAGES) 2>&1 | tee /dev/stderr | $(FAIL_ON_STDOUT)
```
Comment (WangXiangUSTC) [inline Makefile:60]: update go version to 1.12, execute `go tool vet` will return error `Invoking "go tool vet" directly is unsupported; use "go vet"`

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Comment by IANTHEREAL** [comment]
LGTM
